src/components/MembershipTypeButton.tsx

```diff
@@ -0,0 +1,41 @@
+"use client";
+
+import React from "react";
+
+interface MembershipOptionProps {
+  selectedMembership: boolean | undefined;
+  onClick: (isMember: boolean) => void;
+  typeHeading: string;
+  description1: string;
+  description2: string;
+}
+
+export const MembershipTypeButton: React.FC<MembershipOptionProps> = ({
+  selectedMembership,
+  onClick,
+  typeHeading,
+  description1,
+  description2,
+}) => (
+  <label
+    className={`flex flex-col py-[21px] px-8 mb-[24px] h-32 shadow rounded hover:pointer cursor-pointer ${
+      selectedMembership
+        ? "bg-primary text-secondary hover:bg-primary/90 hover:text-secondary"
+        : "bg-secondary hover:bg-primary/90 hover:text-secondary"
```

Comment:
instead of using template literals, use the cn function to merge class names

Reply:
on non-selected, can you make the hover bg-secondary/90 instead

Reply:
Also don't think you need to re-specify this if its not changed.
text should be `text-primary-foreground` / `text-secondary-foreground`
```suggestion
      selectedMembership
        ? "bg-primary text-secondary hover:bg-primary/90"
        : "bg-secondary hover:bg-primary/90"
```